cmd/storj-sim/network.go

```diff
@@ -148,17 +148,22 @@ func newNetwork(flags *Flags) (*Processes, error) {
 	processes := NewProcesses(flags.Directory)
 
 	var (
-		host                   = flags.Host
-		gatewayPort            = 9000
-		bootstrapPort          = 9999
-		bootstrapPrivatePort   = 9988
-		satellitePort          = 10000
-		satellitePrivatePort   = 11000
-		storageNodePort        = 12000
-		storageNodePrivatePort = 13000
-		consolePort            = 10100
-		bootstrapWebPort       = 10010
-		versioncontrolPort     = 10011
+		host                    = flags.Host
+		gatewayPort             = 9000
+		gatewayDebugPort        = 9500
+		bootstrapPort           = 9999
+		bootstrapPrivatePort    = 9988
+		bootstrapDebugPort      = 9977
+		satellitePort           = 10000
+		satellitePrivatePort    = 11000
+		satelliteDebugPort      = 11500
+		storageNodePort         = 12000
+		storageNodePrivatePort  = 13000
+		storageNodeDebugPort    = 13500
+		consolePort             = 10100
+		bootstrapWebPort        = 10010
+		versioncontrolPort      = 10011
+		versioncontrolDebugPort = 10012
```

Comment:
Let's figure out a stable way to set ports to things so it would be easy to figure out them.

Maybe something like:

```
func port(peer, index, subsystem int) int {
    return 10000 + peer * 1000 + index * 10 + port
}
```

where

```
satellite      = 0
storage node   = 1-
uplink         = 5-

single         = 9
   bootstrap      = 0
   versioncontrol = 1

public_grpc  = 0
private_grpc = 1

public_http  = 2

debug_http = 9
```

* a satellite 42 public http port would be `port(satellite, 42, public_http)`, which would be `1-0-42-2`. (this is the console address)
* a storage node 1 debug port would be `port(storage node, 1, debug_http)`, which would be `1-1-01-9`.
* a bootstrap debug port would be `port(single, bootstrap, debug_http)`, which would be `1-9-00-9`.
* uplink 3 public http (gateway) would be `port(uplink, 3, public_http)`, which is `1-5-03-2`

Reply:
`1XZZW`, where `X` is the peer class, `ZZ` is the index and `W` the endpoint.

Reply:
With this notation you are jumping out of the max range of 65539 when running more than 100 of a kind

Reply:
`100 * 10 == 1000`

Reply:
Yeah which means with your notation it ends up like: 1-1-1000-9. Else i dont understand your port collection.

Reply:
```
func port(peerclass, index, endpoint int) int {
    return 10000 + peerclass * 1000 + index * 10 + port
}
```

`port(storage node, 153, 3) = 12533`

Reply:
the index overflows into the peerclass index

Reply:
Oh okay, yeah that works. The Notation was misleading me :)

Reply:
I was also considering using the first number as the peer class, but that would make the ports use across a really wide range. I'm not sure whether that's nice from usability.

```
func portalternate(peerclass, index, endpoint int) int {
    return peerclass*10000 + index * 10 + port
}

satellite      = 0
storage node = 1
uplink         = 2

single         = 3
   // where index is for single is
   bootstrap      = 0
   versioncontrol = 1
```

Reply:
Since most of the people won't be running more than 100 nodes locally anyways, I think it's better to keep the range smaller.